dali/c_api/c_api_test.cc

```diff
@@ -39,81 +39,112 @@ constexpr bool pipelined = true;
-                         .AddArg("output_type", DALI_RGB)
-                         .AddInput("compressed_images", "cpu")
-                         .AddOutput(input_name, "cpu"));
+                             .AddArg("device", execution_device == CPU ? "cpu"s : "mixed"s)
+                             .AddArg("output_type", DALI_RGB)
+                             .AddInput("compressed_images", "cpu")
+                             .AddOutput(input_name, exec_device));
   } else {
     pipe.AddExternalInput(input_name);
   }
   //  Some Op
   pipe.AddOperator(OpSpec("Resize")
-                       .AddArg("device", "cpu")
-                       .AddArg("image_type", DALI_RGB)
-                       .AddArg("resize_x", output_size)
-                       .AddArg("resize_y", output_size)
-                       .AddInput(input_name, "cpu")
-                       .AddOutput("outputs", "cpu"));
+                           .AddArg("device", exec_device)
+                           .AddArg("image_type", DALI_RGB)
+                           .AddArg("resize_x", output_size)
+                           .AddArg("resize_y", output_size)
+                           .AddInput(input_name, exec_device)
+                           .AddOutput("outputs", exec_device));
 
-  vector<std::pair<string, string>> outputs = {{"outputs", output_device}};
+  std::vector<std::pair<std::string, std::string>> outputs = {{"outputs", output_device}};
 
   pipe.SetOutputNames(outputs);
   return pipe_ptr;
 }
 
+
 // Takes Outptus from baseline and handle and compares them
-// Allows only for uint8_t CPU output data to be compared
+// Allows only for uint8_t CPU/GPU output data to be compared
+template<typename Backend>
 void ComparePipelinesOutputs(daliPipelineHandle &handle, Pipeline &baseline) {
   dali::DeviceWorkspace ws;
   baseline.Outputs(&ws);
   daliOutput(&handle);
 
   EXPECT_EQ(daliGetNumOutput(&handle), ws.NumOutput());
   const int num_output = ws.NumOutput();
-  TensorList<CPUBackend> c_output;
   for (int output = 0; output < num_output; output++) {
     EXPECT_EQ(daliNumTensors(&handle, output), batch_size);
     for (int elem = 0; elem < batch_size; elem++) {
       auto *shape = daliShapeAtSample(&handle, output, elem);
       int idx = 0;
-      auto ref_shape = ws.Output<CPUBackend>(output).shape()[idx];
+      auto ref_shape = ws.Output<Backend>(output).shape()[idx];
       for (; shape[idx] != 0; idx++) {
         EXPECT_EQ(shape[idx], ref_shape[idx]);
       }
       EXPECT_EQ(idx, ref_shape.sample_dim());
       free(shape);
     }
 
-    TensorList<CPUBackend> c_output;
-    auto &regular_output = ws.Output<CPUBackend>(0);
-    c_output.Resize(regular_output.shape(), TypeInfo::Create<uint8_t>());
-    daliCopyTensorListNTo(&handle, c_output.raw_mutable_data(), 0, device_type_t::CPU, 0, false);
-    Check(view<uint8_t>(c_output), view<uint8_t>(regular_output));
+    TensorList<CPUBackend> output1, output2;
```

Comment:
Can you give them less generic names, like c_api_out_cpu, regular_our_cpu?

Reply:
Done